### PR TITLE
New image library

### PR DESCRIPTION
**Pull request by blerner**

### CODE REVIEW

File: src/web/js/output-ui.js [hunk 1]
```diff
@@ -178,11 +178,11 @@
       Position.fromPyretSrcloc = function (runtime, srcloc, loc, documents, options) {
         return runtime.ffi.cases(isSrcloc, "Srcloc", loc, {
           "builtin": function(_) {
-             throw new Error("Cannot get Position from builtin location", loc);
+             return new Error("Cannot get Position from builtin location", loc);
```
Comment (jpolitz) [inline src/web/js/output-ui.js]: These seem like weird changes to have in the images update.

File: package.json [hunk 2]
```diff
@@ -43,6 +43,7 @@
     "karma-sourcemap-loader": "0.3.7",
     "karma-webpack": "^3.0.5",
     "load-script": "^1.0.0",
+    "lodash": "~2.4.1",
```
Comment (jpolitz) [inline package.json]: Where is lodash used?

File: src/web/js/output-ui.js (hunk 1 above)
Comment (blerner) [reply src/web/js/output-ui.js]: I mostly agree.  However, IIRC the commit that introduced that (https://github.com/brownplt/code.pyret.org/commit/1eced4f62b1c15e35b20bf93b3d8893b4231a80f) was trying to recover from some weird stack issue that triggered in deeply-nested images when rendered in error messages, or something like that, and it boiled down to fixing the stack-unsafety of this case.  This change should be cherry-picked anyway, regardless of this PR.

**Comment by blerner** [comment]
No idea. I don't even remember editing package.json...

**Comment by jpolitz** [comment]
- [ ] Make sure type and untyped provide the same functions
- [ ] Check naming of new files for import/include back-compat
- [ ] Check on needing to import image-structs

**Pull request merged**